app/services/hearing_result_fetcher.rb

```diff
@@ -14,23 +14,33 @@ def call
       sitting_day: sitting_day,
     )
 
-    msg = "[#{Current.request_id}] - "
+    if response.success?
+      if response.body.present?
+        HearingsCreator.call(
+          hearing_resulted_data: response.body.deep_stringify_keys,
+          queue_url: Rails.configuration.x.aws.sqs_url_hearing_resulted,
+        )
+      else
+        msg = log_for("Past result for hearing ID #{hearing_id} is not available.")
+        Rails.logger.info(msg)
+      end
+    end
 
-    if response.success? && response.body.blank?
-      msg += "Past result for hearing ID #{hearing_id} is not available."
+    if (300..499).cover?(response.status)
+      msg = log_for("Unable to fetch past result of hearing ID #{hearing_id}: Common Platform responded with status code #{response.status}.")
       Rails.logger.info(msg)
-      raise StandardError, msg
     end
 
-    unless response.success?
-      msg += "Unable to fetch past result of hearing ID #{hearing_id}: Common Platform responded with status code #{response.status}."
-      Rails.logger.info(msg)
-      raise StandardError, msg
+    if (500..599).cover?(response.status) # Server Error 5XX
+      msg = log_for("Unable to fetch past result of hearing ID #{hearing_id}: Common Platform responded with status code #{response.status}.")
+
+      raise Faraday::ServerError, msg # this tells Sidekiq to Retry
```

Comment:
it retries the hearing fetching only in case of server error, which is most of the time 503